elastic/kibana
# [Synthetics UI] Monitor Status Panel (histogram) - Summary / History Page

Closes #141903

## Summary

This PR adds the Monitor **Status** panel which consists of a histogram chart, on Synthetics -> Go to _a Monitor_ -> Monitor Summary | Monitor History pages. The histogram represents monitor Failed or Complete runs over time, with variation of the colors representing monitor Availability.

Chart maintains a minimum width of the time slices to not let slots reduce to faint lines when a large set of data is plotted. If window resizes, it'll recalculate the slice time range based on the available width.

The same component is used on both Monitor Summary and Monitor History pages, with the following changes:
- On Monitor Summary page, the time bounds will always be the last 24 hours and the chart is not brushable/zoomable.
- On Monitor History page, the time bounds are the ones selected in the app's Time Selector  (SyntheticsDatePicker). Chart is brushable on this page and brushing the chart will update the time range in the Time Selector.

### Notes
- **Explore** button as shown in the designs is not implemented due to limitations in Exploratory View/Lens.
- The Error Annotations (red warning icons shown in the design) are not implemented in this PR due to limitations in @elastic/charts' Heatmap chart.

### How to test
1. Setup Kibana so that you are able to create or push synthetics monitors.
2. In Kibana -> Advanced Settings, enable Synthetics.
3. Create or push a few monitors with relatively high frequencies so that you can test a few monitor runs plotted on the chart.
4. Wait for the documents to come back.
5. Goto Synthetics ->  _From ... menu of a monitor select_ "Go to Monitor".
6. Observe the **Status** panel and make sure you see the failed or completed test runs plotted appropriately. You can check the **Last 10 Test Runs** to confirm if data has been ingested.
7. Click **View History** and make sure it navigates to the History page.
8. On the History page, brush the chart to zoom in on a set of test runs and make sure that it changes the time in the Time Selector accordingly.
9. Change the time in the Time Selector and observe that the chart plots the data per the specified time range.

_Note_: You can also use this [deployment](https://p.elstc.co/paste/-tV0+10z#X4m1legX1b512lO6ow9vSmLd9ktwmHU6BQ4hx2vQpHP) to test (if it's still live).

**Video**

https://user-images.githubusercontent.com/2748376/199543972-ac408c55-9d64-4960-b734-c9357878af51.mov

**Screenshots**
<img width="2560" alt="Screenshot 2022-11-02 at 17 12 54" src="https://user-images.githubusercontent.com/2748376/199543544-15cf1d44-e322-4a18-85f4-1caea84a69ff.png">

![Screenshot 2022-11-02 at 17 06 32](https://user-images.githubusercontent.com/2748376/199543741-e0427d58-0983-458f-8447-9c2221bf5530.png)
<img width="1776" alt="Screenshot 2022-11-02 at 17 07 29" src="https://user-images.githubusercontent.com/2748376/199543837-ed60f823-3fa6-4f73-a557-b747cb3ccb2f.png">

<img src="https://user-images.githubusercontent.com/2748376/199664308-15250493-836f-4e51-b9dd-0c0b48d8c1b1.png" width="390">
